tests/network/bgp/conftest.py

```diff
@@ -0,0 +1,175 @@
+import os
+from collections.abc import Generator
+from pathlib import Path
+from typing import Final
+
+import ocp_resources.network_config_openshift_io as openshift_nc
+import pytest
+from ocp_resources.config_map import ConfigMap
+from ocp_resources.namespace import Namespace
+from ocp_resources.node import Node
+from ocp_resources.pod import Pod
+
+from libs.net import netattachdef as libnad
+from libs.net.bgp import (
+    create_cudn_route_advertisements,
+    create_frr_configuration,
+    deploy_external_frr_pod,
+    enable_route_advertisements_in_cluster,
+    generate_frr_conf,
+)
+from libs.net.node_network import DEFAULT_ROUTE_V4, lookup_br_ex_gateway_v4
+from libs.net.udn import create_udn_namespace
+from tests.network.libs import cluster_user_defined_network as libcudn
+from tests.network.libs import nodenetworkconfigurationpolicy as libnncp
+from tests.network.libs.label_selector import LabelSelector
+from utilities.infra import cache_admin_client, get_node_selector_dict
+
+APP_CUDN_LABEL: Final[dict] = {"app": "cudn"}
+BGP_DATA_PATH: Final[Path] = Path(__file__).resolve().parent / "data" / "frr-config"
+CUDN_BGP_LABEL: Final[dict] = {"cudn-bgp": "blue"}
+CUDN_SUBNET_IPV4: Final[str] = "192.168.10.0/24"
+EXTERNAL_SUBNET_IPV4: Final[str] = "172.100.0.0/16"
+
+
+@pytest.fixture(scope="module")
+def vlan_nncp(vlan_base_iface: str, worker_node1: Node) -> Generator[libnncp.NodeNetworkConfigurationPolicy]:
+    """Creates a NodeNetworkConfigurationPolicy with a VLAN interface on the specified cluster's node."""
+    with libnncp.NodeNetworkConfigurationPolicy(
+        name="test-vlan-nncp",
+        desired_state=libnncp.DesiredState(
+            interfaces=[
+                libnncp.Interface(
+                    name=f"{vlan_base_iface}.{os.environ['VLAN_TAG']}",
+                    state=libnncp.NodeNetworkConfigurationPolicy.Interface.State.UP,
+                    type="vlan",
+                    vlan=libnncp.Vlan(id=int(os.environ["VLAN_TAG"]), base_iface=vlan_base_iface),
+                )
+            ]
+        ),
+        node_selector=get_node_selector_dict(node_selector=worker_node1.hostname),
+    ) as nncp:
+        nncp.wait_for_status_success()
+        yield nncp
+
+
+@pytest.fixture(scope="session")
+def br_ex_gateway_v4(worker_node1: Node) -> str:
+    """Retrieves the IPv4 gateway address of the external bridge on the specified worker node."""
```

Comment:
the fixture name is descriptive enough; this can be dropped.
this applies to all relevant places - names should be descriptive so no need for docsring just to repeat what the name does.
in places where docstring is needed, proper google-format is to be used